apache/flink
# [FLINK-3551] [examples] Sync Scala Streaming Examples

Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

- Add IterateExample to iteration example
- Add ReadFromKafka to Kafka example
- Add WriteIntoKafka to Kafka example
- Add IncrementalLearningSkeleton to ml example
- Add TwitterExample to Twitter example
- Add GroupedProcessingTimeWindowExample to windowing example
- Add SessionWindowing to windowing example
- Add WindowWordCount to windowing example
- Add WordCount to wordcount example
